renderite/serialization.zig

```diff
@@ -30,8 +30,8 @@ pub const IpcDeserializer = struct {
             .bool => return try self.readBool(),
             .@"enum" => return try self.readEnum(T),
             .optional => return try self.readNullable(T),
-            // else => @compileError(std.fmt.comptimePrint("Unsupported type {s} for deserialization", .{@typeName(T)})),
-            else => return error.TypeNotSupported,
+            else => @compileError(std.fmt.comptimePrint("Unsupported type {s} for deserialization", .{@typeName(T)})),
+            // else => return error.TypeNotSupported,
```

Comment:
You can just omit the `else` branch and unhandled cases will automatically be a compile error, Zig does not automatically place an empty `else` branch like C#

```suggestion
            // else => return error.TypeNotSupported,
```

Reply:
This is intentional, if I see this error I know exactly what it means and where it happened.

Reply:
i see